crypto/ring-signature/src/ring_signature/mlsag.rs

```diff
@@ -124,114 +118,46 @@ impl RingMLSAG {
         check_value_is_preserved: bool,
         // Note: this `mut rng` can just be `rng` if this is merged upstream:
         // https://github.com/dalek-cryptography/curve25519-dalek/pull/394
-        mut rng: &mut dyn CryptoRngCore,
+        rng: &mut dyn CryptoRngCore,
     ) -> Result<Self, Error> {
         let ring_size = ring.len();
 
-        if real_index >= ring_size {
-            return Err(Error::IndexOutOfBounds);
-        }
-
-        let G = B_BLINDING;
-        debug_assert!(
-            generator.B_blinding == G,
-            "basepoint for blindings mismatch"
+        // Setup buffers
+        let (mut responses, mut decompressed_ring) = (
+            alloc::vec![CurveScalar::from(Scalar::zero()); 2 * ring_size],
+            alloc::vec![(RistrettoPublic::default(), Commitment::default()); ring_size],
```

Comment:
Is the idea that the buffers would get set up off of the ledger, and then the signing of each ring happens individually on the ledger to fill the buffers?

Reply:
not sure what you mean by buffers? the streaming is to mitigate packet size constraints and avoid holding these (challenge/response and ring entry) buffers on the ledger when constructing each ring signature.

the process from the devices perspective (ie. APDUs over USB) is documented [here](https://mobilecoinofficial.github.io/ledger-mob/ledger_mob_core/#executing-a-transaction) (which matches a [state machine](https://mobilecoinofficial.github.io/ledger-mob/src/ledger_mob_core/engine/ring.rs.html#118) on the device), but from a host perspective it's just an implementation of [`RingSigner`](https://mobilecoinofficial.github.io/ledger-mob/ledger_mob/tx/struct.TransactionHandle.html#impl-RingSigner-for-TransactionHandle%3CT%3E) that manages this [internally](https://mobilecoinofficial.github.io/ledger-mob/src/ledger_mob/tx/ring.rs.html#61).